client/blocks/taxonomy-manager/list-item.jsx

```diff
@@ -56,8 +58,11 @@ class TaxonomyManagerListItem extends Component {
 
 	closeDeleteDialog = action => {
 		if ( action === 'delete' ) {
-			const { siteId, taxonomy, term } = this.props;
-			this.props.deleteTerm( siteId, taxonomy, term.ID, term.slug );
+			const { siteId, taxonomy, term, afterDelete } = this.props;
+			this.props.deleteTerm( siteId, taxonomy, term.ID, term.slug )
+				.then( () => {
+					afterDelete( term );
+				} );
```

Comment:
One of my other ideas that I deemed hack_ier_ was to do something in the reducer logic to increment the post_count.  I recall us doing something similar with post counts in the post listing - so perhaps that is the best way to handle this.